### PR TITLE
Add docker epoch image builds

### PR DESCRIPTION
**Pull request by dincho**
[PT 155067543](https://www.pivotaltracker.com/story/show/155067543)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -315,6 +350,13 @@ workflows:
             tags:
               only: /^v.*$/
 
+      - docker_image_build:
+          filters:
+            branches:
```
Comment (lucafavatella) [inline .circleci/config.yml]: If slow or we run out of containers, we may run this only on master.

File: .circleci/config.yml [hunk 2]
```diff
@@ -8,6 +8,29 @@ references:
...
+        curl -L -o /tmp/docker-${DOCKER_CLIENT_VERSION:?}.tgz \
+          https://download.docker.com/linux/static/stable/x86_64/docker-${DOCKER_CLIENT_VERSION:?}.tgz
+        tar -xz -C /tmp -f /tmp/docker-${DOCKER_CLIENT_VERSION:?}.tgz
+        mkdir ~/bin && mv /tmp/docker/docker ~/bin
+
+  docker_build: &docker_build
+    run:
+      name: Build Docker image
+      command: |
+        ~/bin/docker build -t $DOCKERHUB_REPO:$DOCKER_TAG .
```
Comment (lucafavatella) [inline .circleci/config.yml]: Why the need for tagging it at all? No uploading is done so it looks like no tagging needed.

Also, I wonder if any state is persisted between runs so there is clash in tag name? (that BTW is branch name)

File: .circleci/config.yml (hunk 1 above)
Comment (dincho) [reply .circleci/config.yml]: It's about 11minutes.

File: .circleci/config.yml (hunk 2 above)
Comment (dincho) [reply .circleci/config.yml]: I'll tag and push in the following pull requests, anyway I removed the obsoleted tagging (in scope of this PR) in 0d1f9069 for the sake of atomicity

File: .circleci/config.yml (hunk 1 above)
Comment (lucafavatella) [reply .circleci/config.yml]: So maybe worth running it only on master?

File: .circleci/config.yml (hunk 2 above)
Comment (lucafavatella) [reply .circleci/config.yml]: thanks

File: .circleci/config.yml (hunk 1 above)
Comment (dincho) [reply .circleci/config.yml]: I don't mind if we setup proper monitoring on master failure and development policy to fix it with highest priority. But let's do that in future, once we agree on policy and have proper monitoring.

**Pull request merged**